src/lib/server/searchWeb.ts

```diff
@@ -0,0 +1,34 @@
+import { SERPAPI_KEY } from "$env/static/private";
+import type { Message } from "$lib/types/Message";
+
+import { getJson } from "serpapi";
+import type { GoogleParameters } from "serpapi";
+import { generateFromDefaultEndpoint } from "./generateFromDefaultEndpoint";
+
+// Show result as JSON
+export async function searchWeb(query: string) {
+	const params = {
+		q: query,
+		hl: "en",
+		gl: "us",
+		google_domain: "google.com",
+		api_key: SERPAPI_KEY,
+	} satisfies GoogleParameters;
+
+	// Show result as JSON
+	const response = await getJson("google", params);
+
+	return response;
+}
+
+export async function getQueryFromPrompt(messages: Pick<Message, "from" | "content">[]) {
+	let prompt = "What query would you ask Google to answer this question?\n\n";
+
+	messages.forEach((message) => {
+		prompt += `${message.from}: ${message.content}\n`;
+	});
+
+	prompt += "\nQuery:";
+
+	return await generateFromDefaultEndpoint(prompt);
```

Comment:
I don't know if it's intended but it seems like when you have more than one message, the prompt sent to get the Google query is including the assistant answer. So it seems like it gets confused. I have been testing with this example:
- What is the capital of Spain?
- What are the colors of the football squad?

And the prompt is something like:
```
What query would you ask Google to answer this question?

user: what is the capital of Spain?
assistant:  The current capital of Spain since the year 1916 is Madrid. This was established by the law of 10/04/1617 issued by Felipe III. However before this time, other Spanish kings chose Segovia, Toledo, Valladolid & Cordoba to exercise their rule as well as residences such as El Pardo
user: what are the colors of the football squad?

Query:
```

Reply:
I changed it so that it only uses the user prompts to formulate the web query. Might make it easier to get a good query out.